tests/integration/hpe_3par_manager.py

```diff
@@ -180,6 +183,64 @@ def hpe_inspect_volume(self, volume, **kwargs):
 
         return inspect_volume
 
+    def hpe_inspect_share(self, volume, **kwargs):
+        inspect_volume = self.client.inspect_volume(volume['Name'])
+        self.assertEqual(volume['Name'], inspect_volume['Name'])
+        self.assertEqual(volume['Driver'], inspect_volume['Driver'])
+        self.assertEqual(volume['Options'], inspect_volume['Options'])
+        self.assertIn('Status', inspect_volume)
+        timeout = time.time() + 600
+        while 1:
+            inspect_volume = self.client.inspect_volume(volume['Name'])
+            time.sleep(10)
+            if inspect_volume['Status']['status'] == 'AVAILABLE' or time.time() > timeout:
+                break
+
+        if inspect_volume['Status']['status'] == 'AVAILABLE':
+            # def hpe_inspect_share(self, volume, **kwargs):
+            #inspect_volume = self.client.inspect_volume(volume['Name'])
+            #status = ['backend', 'cpg', 'fpg', 'name', 'protocol', 'size', 'status', 'vfs', 'vfsIPs']
+            hpe3par_cli = self._hpe_get_3par_client_login_filePersona()
+            #uri = '/fileshares/%s' % inspect_volume['Status']['id']
+            uri = '/fileshares/'
+            fileshare_list = hpe3par_cli.http.get(uri)
+            members = fileshare_list[1]['members']
+            for i in members:
+                if i.get('name') == volume['Name']:
+                    fshare_index = members.index(i)
+                    fshare_details = members[fshare_index]          
+            uri = '/fileshares/%s' % fshare_details['id']
+            fileshare_info = hpe3par_cli.http.get(uri)
+
+            self.assertEqual(fileshare_info[1]['name'], inspect_volume['Name'])
+            self.assertEqual(inspect_volume['Status']['protocol'], 'nfs')
+            self.assertEqual(inspect_volume['Status']['fpg'], fileshare_info[1]['fpg'])
+            self.assertEqual(inspect_volume['Status']['vfs'], fileshare_info[1]['vfs'])
+            show_fpg = hpe3par_cli._run(['showfpg', '-d', fileshare_info[1]['fpg']])
+            self.assertEqual(inspect_volume['Status']['cpg'], show_fpg[17].split(':')[1].strip())

```

Comment:
'showfpg' CLI returns a list and hence used indexing. Since the CLI output will remain constant there is low possibility of the indexing changing.

Output snippet.
display show_fpg: ['------------------File Provisioning Group---------------------', 'File Provisioning Group :  Test_Fpg_2', 'Active path             :  /Test_Fpg_2', 'Active State            :  ACTIVATED', 'Freeze State            :  NOT_FROZEN', 'Isolation State         :  ACCESSIBLE', 'Upgrade State           :  OK', 'Version                 :  12.3', 'FsGeneration            :  1', 'UUID                    :  671a6468-383c-4eea-aa6f-8d139bf50ae4', 'Filesystem Number       :  10', 'Size (GiB)              :  16384.00', 'Free (GiB)              :  16375.54', 'Available (GiB)         :  16375.54', 'Used (GiB)              :  8.46', 'Files                   :  58', 'Files Free              :  35473710571', 'Default CPG             :  swap_fs_cpg', 'VVs                     :  Test_Fpg_2.1', 'Primary Node            :  1', 'Alternate Node          :  0', 'Current Node            :  1', 'Comment                 :  Docker created FPG', 'State                   :  normal', 'Health Description      :  FPG is activated and operational', 'Corrective Action       :  --', 'RC Group                :  --', '', 'SegmentNumber,FSCKState,FSCKPhaseRequired', '1,NOT_REQUIRED,NONE', '', 'Domain,Owner,FsName,Filesets,VVIDs,Nodes,IpFsType', 'bf88e65b-259f-4b52-aa5d-28a1521e121f,1,Test_Fpg_2,fileset1,12941,,ADE', '', 'Volume,VVID,Nodes,Capacity(GiB)', 'Test_Fpg_2.1,12941,"1,0",16384.00', '']